core/src/event_id.rs

```diff
@@ -96,49 +122,92 @@ impl EventId {
             return None; // not a CIP-124 EventID
         };
 
-        let (_network_id, mut remainder) = de_varint(remainder).unwrap_or_default();
+        let (network_id, mut remainder) = de_varint(remainder).unwrap_or_default();
+
+        // separator [u8; 8]
+        let separator = &remainder[..8];
+        // controller [u8; 8]
+        let controller = &remainder[8..8 + 8];
+        // StreamID [u8; 4]
+        let stream_id = &remainder[8 + 8..8 + 8 + 4];
 
-        // strip separator [u8; 8] controller [u8; 8] StreamID [u8; 4]
         remainder = &remainder[(8 + 8 + 4)..];
 
         // height cbor unsigned integer
-        if remainder[0] <= 23 {
+        let (height, cid) = if remainder[0] <= 23 {
```

Comment:
It might be worth extracting this to a small function, or at least adding a comment about why we're using big and little endian bytes. I had to look up how CBOR integer encoding is implemented to understand this.

Reply:
I had too read the spec as well to update the code. I'll add a comment and refactor into a function.